test/backend/CMakeLists.txt

```diff
@@ -1,15 +1,42 @@
 # SPDX-License-Identifier: Apache-2.0
 
-# Set TEST_DRIVER_BUILD_PATH to the directory of the test driver.
-# It is expected to have two sub-directories:
-# - 'bin' containing executable files, and
-# - 'lib' containing necessary libraries for runtime execution.
-set (TEST_DRIVER_BUILD_PATH ${CMAKE_BINARY_DIR})
-# Set TEST_DRIVER_COMMAND to the command used to compile models.
-# The command is expected to be found in the 'bin' directory of ${TEST_DRIVER_BUILD_PATH}
-set (TEST_DRIVER_COMMAND "onnx-mlir")
-configure_file(test.py test.py COPYONLY)
-configure_file(test_config.py.in test_config.py)
+if (NOT "${CMAKE_CFG_INTDIR}" STREQUAL ".")
+  foreach(ONNX_MLIR_BUILD_MODE ${CMAKE_CONFIGURATION_TYPES})
+    string(REPLACE ${CMAKE_CFG_INTDIR} ${ONNX_MLIR_BUILD_MODE}
+      ONNX_MLIR_LIBRARY_PATH_CONFIGURED
+      ${ONNX_MLIR_LIBRARY_PATH})
+    string(REPLACE ${CMAKE_CFG_INTDIR} ${ONNX_MLIR_BUILD_MODE}
+      ONNX_MLIR_RUNTIME_PATH_CONFIGURED
+      ${ONNX_MLIR_RUNTIME_PATH})
+
+    configure_file(
+      ${CMAKE_CURRENT_SOURCE_DIR}/test.py
+      ${CMAKE_CURRENT_BINARY_DIR}/${ONNX_MLIR_BUILD_MODE}/test.py
+      COPYONLY
+      )
+
+    configure_file(
+      ${CMAKE_CURRENT_SOURCE_DIR}/test_config.py.in
+      ${CMAKE_CURRENT_BINARY_DIR}/${ONNX_MLIR_BUILD_MODE}/test_config.py
+      @ONLY
+      )
+  endforeach()
+else()
+  set(ONNX_MLIR_LIBRARY_PATH_CONFIGURED ${ONNX_MLIR_LIBRARY_PATH})
+  set(ONNX_MLIR_RUNTIME_PATH_CONFIGURED ${ONNX_MLIR_RUNTIME_PATH})
+
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/test.py
+    ${CMAKE_CURRENT_BINARY_DIR}/test.py
+    COPYONLY
+    )
+
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/test_config.py.in
+    ${CMAKE_CURRENT_BINARY_DIR}/test_config.py
+    @ONLY
+    )
+endif()
```

Comment:
Can we have some comments for this block of code? e.g., when does `CMAKE_CFG_INTDIR` resolve to a subdirectory and when does it resolve to a "."? What's the purpose of the `string(REPLACE ...)`?

Reply:
I originally pushed an update which adds some more details to explain the block of code, but after spending some more time thinking through the problem, it is possible to simplify the logic and use generator expressions more heavily, so the latest change does just that.